### PR TITLE
747 advanced search interaction label predicates

### PR DESCRIPTION
**Pull request by brianbolt**
Only add label join if label predicates provided in the search criteria

### CODE REVIEW

File: src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java [hunk 1]
```diff
@@ -2389,50 +2386,56 @@ else if (query.getRecordedDateLessThan() != null){
...
+							}else if(interaction.getOperator().equals("<=")){
+								Predicate firstThingLabelLessThan = criteriaBuilder.lessThanOrEqualTo(firstThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								firstItxPredicates.add(firstThingLabelLessThan);
+							}else{
+								Predicate firstThingLabelLike = criteriaBuilder.like(criteriaBuilder.lower(firstThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+								firstItxPredicates.add(firstThingLabelLike);
+							}
 						}else{
-							Predicate firstThingLabelLike = criteriaBuilder.like(criteriaBuilder.lower(firstThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+							Predicate firstThingLabelLike = criteriaBuilder.like(firstThingLabel.<String>get("labelText"), '%' + interaction.getThingLabelText() + '%');
```
Comment (bffrost) [inline src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java:2436]: Just noticed here that the analogous query above is case-insensitive but this one is case-sensitive. I think it'd make sense for this default "no operator specified" to be case-insensitive too.

File: src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java [hunk 2]
```diff
@@ -2474,50 +2474,56 @@ else if (query.getRecordedDateLessThan() != null){
...
+								Predicate secondThingLabelGreaterThan = criteriaBuilder.greaterThanOrEqualTo(secondThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								secondItxPredicates.add(secondThingLabelGreaterThan);
+							}else if(interaction.getOperator().equals("<")){
+								Predicate secondThingLabelLessThan = criteriaBuilder.lessThan(secondThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								secondItxPredicates.add(secondThingLabelLessThan);
+							}else if(interaction.getOperator().equals("<=")){
+								Predicate secondThingLabelLessThan = criteriaBuilder.lessThanOrEqualTo(secondThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								secondItxPredicates.add(secondThingLabelLessThan);
+							}else{
+								Predicate secondThingLabelLike = criteriaBuilder.like(secondThingLabel.<String>get("labelText"), '%' + interaction.getThingLabelText() + '%');
```
Comment (bffrost) [inline src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java:2520]: Same thing here. Should make this case insensitive IMO.

**Review by bffrost — APPROVED** [review]
Left two comments that are optional to resolve. I am recommending a behavior change towards what I think is better default behavior, but feel free to opt not to change them.

File: src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java (hunk 1 above)
Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java:2436]: Yes, I will make this change.

File: src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java (hunk 2 above)
Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java:2520]: Interesting that its the opposite here. fixing to be the same (both lower)

**Pull request merged**